evergreen/eg_app/views.py

```diff
@@ -12,6 +20,61 @@
 def index(request):
     return render(request,'index.html')
 
+def getFileType(filePath) -> tuple[str|None, str|None]:
+    contentType = mimetypes.guess_type(filePath) #returns tuple {type, encoding}
+    return contentType
+
+def fileHandler(request: HttpRequest, fileName: str) -> HttpResponse:  # can handle img and text
+    # get the absolute path
+    sanitizedFileName = quote(fileName)
+    path = Path(settings.STATIC_ROOT) / sanitizedFileName
+
+    allowedType: Set[str] = {'.css', '.html', '.js', '.png', '.jpg', '.jpeg', '.gif', '.mp3', '.mp4', '.xml', '.json', '.pdf', '.ico'}  # can add more
+
+    if not str(path.suffix.lower()) in allowedType:  # to deal with user uploads
+        return HttpResponseNotFound("404 - File type not allowed")
+
+    # deal with /../ attacks
+    rootPath = Path(settings.STATIC_ROOT).resolve()
+    if not path.resolve().is_relative_to(rootPath):
+        return HttpResponseNotFound("404 Not Found")
+
+    if path.exists() and path.is_file(): # make sure it's not a directory
+        contentType, encoding = getFileType(str(path))
+
+        try:
+            with open(path, 'rb') as file:
+                content = file.read()
+
+            if contentType is not None:
+                response = HttpResponse(content, content_type=contentType)
+            else:
+                response = HttpResponse(content)
+
+            if encoding is not None:
+                response['Content-Encoding'] = encoding
+            else:
+                response['Content-Encoding'] = ""
+
+            response['X-Content-Type-Options'] = "nosniff"
+            # response['Content-Length'] = str(len(content))  # needed for very large files, Django handles it
+
+            return response
+
+        except OSError:  # catch most of them, like FileNotFoundError
+            return HttpResponseNotFound("404 Not Found")
+            # return render(request, '404.html', status=404) when making 404 pages
+    else:
+        return HttpResponseNotFound("404 Not Found")
+
+
+def addCookies(response, cookies): # add all cookies from the give dic to the response
```

Comment:
One more set of type hints (and make it a docstring comment)
```suggestion
def addCookies(response: HttpResponse, cookies: dict[str, str]): 
    """ Add all cookies from the given dict to the response """
```